File: client-domain/src/main/java/com/jraska/github/client/rx/AppSchedulers.java
package com.jraska.github.client.rx;

import io.reactivex.Scheduler;

public final class AppSchedulers {
  private final Scheduler mainThread;
  private final Scheduler io;
  private final Scheduler computation;

  public AppSchedulers(Scheduler mainThread, Scheduler io, Scheduler computation) {
    this.mainThread = mainThread;
    this.io = io;
    this.computation = computation;
  }

  public Scheduler mainThread() {
    return mainThread;
  }

  public Scheduler io() {
    return io;
  }

  public Scheduler computation() {
    return computation;
  }
}

File: core/src/main/java/com/jraska/github/client/rx/AppSchedulers.kt
package com.jraska.github.client.rx

import io.reactivex.Scheduler

class AppSchedulers(val mainThread: Scheduler, val io: Scheduler, val computation: Scheduler) {

  fun mainThread(): Scheduler {
    return mainThread
  }

  fun io(): Scheduler {
    return io
  }

  fun computation(): Scheduler {
    return computation
  }
}
